tests/circ_print_test.py

```diff
@@ -0,0 +1,31 @@
+from io import StringIO
+import unittest
+from unittest import TestCase
+from unittest.mock import patch
+import qforte as qf
+from qforte import *
+
+class CircPrintTest(TestCase):
+    def test_gate_print(self):
+
+        # initialize empty circuit
+        circ = qf.Circuit()
+        # add (Z1 H2 Y4 X4) Pauli string
+        # note: the rightmost gate is applied first
+        circ.add(qf.gate('X', 4))
+        circ.add(qf.gate('Y', 4))
+        circ.add(qf.gate('H', 2))
+        circ.add(qf.gate('Z', 1))
+
+        # test built-in print function
+        with patch('sys.stdout', new = StringIO()) as fake_out:
+            print(circ)
+            self.assertEqual(fake_out.getvalue(), '[Z1 H2 Y4 X4]\n')
+
+        # test smart_print function
+        with patch('sys.stdout', new = StringIO()) as fake_out:
```

Comment:
@imagoulas and @nstair: I had these two suggested changes pending and so they might have not be reported in the PR comments. It would be nice to apply these simplifications to the test case that was committed in a new smaller PR. Thanks!

Reply:
@fevangelista : Before I proceed with the suggested changes, I would like a clarification. At the moment, the test is checking the output of both the built-in print function as well as the smart_print function (i.e., we are making sure that the user will see in their screen exactly what we wanted them to see). If I understand this correctly, with the suggested change, instead of checking both print and smart_print,  we will be checking only that the order of the operators is correct in str(circ). If you could confirm that this is correct, I will replace the tests for print and smart_print by a single test of str(circ).

Reply:
You are right, with this change we could check only the result of print. But I think that we want to make smart_print obsolete in the long run, or call it something like `show`. Perhaps right now it is enough to just change the part that pertains to print.